simpeg_drivers/potential_fields/gravity/params.py

```diff
@@ -11,309 +11,94 @@
-
-    @gx_channel.setter
-    def gx_channel(self, val):
-        self.setter_validator("gx_channel", val, fun=self._uuid_promoter)
-
-    @property
-    def gx_uncertainty(self):
-        return self._gx_uncertainty
-
-    @gx_uncertainty.setter
-    def gx_uncertainty(self, val):
-        self.setter_validator("gx_uncertainty", val, fun=self._uuid_promoter)
-
-    @property
-    def gy_channel_bool(self):
-        return self._gy_channel_bool
-
-    @gy_channel_bool.setter
-    def gy_channel_bool(self, val):
-        self.setter_validator("gy_channel_bool", val)
-
-    @property
-    def gy_channel(self):
-        return self._gy_channel
-
-    @gy_channel.setter
-    def gy_channel(self, val):
-        self.setter_validator("gy_channel", val, fun=self._uuid_promoter)
-
-    @property
-    def gy_uncertainty(self):
-        return self._gy_uncertainty
+class GravityInversionParams(BaseInversionData):
+    """
+    Gravity inversion parameters.
+
+    :param gx_channel: gx channel.
+    :param gy_channel: gy channel.
+    :param gz_channel: gz channel.
+    :param gxx_channel: gxx channel.
+    :param gxy_channel: gxy channel.
+    :param gxz_channel: gxz channel.
+    :param gyy_channel: gyy channel.
+    :param gyz_channel: gyz channel.
+    :param gzz_channel: gzz channel.
+    :param gx_uncertainty: gx uncertainty.
+    :param gy_uncertainty: gy uncertainty.
+    :param gz_uncertainty: gz uncertainty.
+    :param gxx_uncertainty: gxx uncertainty.
+    :param gxy_uncertainty: gxy uncertainty.
+    :param gxz_uncertainty: gxz uncertainty.
+    :param gyy_uncertainty: gyy uncertainty.
+    :param gyz_uncertainty: gyz uncertainty.
+    :param gzz_uncertainty: gzz uncertainty.
+    """
 
-    @gy_uncertainty.setter
-    def gy_uncertainty(self, val):
-        self.setter_validator("gy_uncertainty", val, fun=self._uuid_promoter)
+    name: ClassVar[str] = "Gravity Inversion"
+    title: ClassVar[str] = "Gravity Inversion"
+    default_ui_json: ClassVar[Path] = assets_path() / "uijson/gravity_inversion.ui.json"
+
+    inversion_type: str = "gravity"
+    physical_property: str = "density"
+
+    gx_channel: FloatData | None = None
+    gy_channel: FloatData | None = None
+    gz_channel: FloatData | None = None
+    gxx_channel: FloatData | None = None
+    gxy_channel: FloatData | None = None
+    gxz_channel: FloatData | None = None
+    gyy_channel: FloatData | None = None
+    gyz_channel: FloatData | None = None
+    gzz_channel: FloatData | None = None
+    gx_uncertainty: FloatData | float | None = None
+    gy_uncertainty: FloatData | float | None = None
+    gz_uncertainty: FloatData | float | None = None
+    gxx_uncertainty: FloatData | float | None = None
+    gxy_uncertainty: FloatData | float | None = None
+    gxz_uncertainty: FloatData | float | None = None
+    gyy_uncertainty: FloatData | float | None = None
+    gyz_uncertainty: FloatData | float | None = None
+    gzz_uncertainty: FloatData | float | None = None
```

Comment:
This is where we could group the channel and its uncertainty in a "component" block. It's then pretty generic to loop over components, and only fetch those that were set.

Reply:
This is a nice idea, but we would need a special handler inside BaseData.collect_input_from_dict that locates the groups and formats an entry into the update dictionary for this to work.

Reply:
I don't think so? Let's talk.

Reply:
From our discussion.  To be addressed in another issue.  @domfournier will look into using an alias to allow existing BaseData.collect_input_from_dict definition to properly pull data from ui.json and store in DataComponent class instances.